core/src/main/resources/error/error-classes.json

```diff
@@ -1009,6 +1009,11 @@
           "<value>."
         ]
       },
+      "REGEX_GROUP_INDEX_EXCEED_REGEX_GROUP_COUNT" : {
+        "message" : [
+          "Regex group count is <groupCount>, but the specified group index is <groupIndex>."
```

Comment:
Not obvious the valid range for `groupIndex`. Could you write the message in the format:
```
expects ..., but got ...
```